### PR TITLE
feat(logging): add structured fields in logging + new log format

### PR DESCRIPTION
**Pull request by gurinderu**
## Description

Added particle_id label for the structured logging 
Added configuration for the log format

### CODE REVIEW

File: aquamarine/src/particle_functions.rs [hunk 1]
```diff
@@ -198,7 +198,8 @@ impl<F: ParticleFunctionStatic> Functions<F> {
             };
 
             if let Err(err) = &result {
-                log::warn!(
+                tracing::warn!(
+                    particle_id = particle_id,
                     "Failed host call {} ({}) [{}]: {}",
```
Comment (kmd-fl) [inline aquamarine/src/particle_functions.rs:203]: mb then remove particle_id from the message?

File: connection-pool/src/behaviour.rs [hunk 2]
```diff
@@ -214,17 +214,17 @@ impl ConnectionPoolBehaviour {
             outlet.send(SendStatus::Ok).ok();
             self.wake();
         } else if self.contacts.contains_key(&to.peer_id) {
-            log::debug!(target: "network", "{}: Sending particle {} to {}", self.peer_id, particle.id, to.peer_id);
+            tracing::debug!(target: "network",particle_id = particle.id , "{}: Sending particle {} to {}", self.peer_id, particle.id, to.peer_id);
```
Comment (kmd-fl) [inline connection-pool/src/behaviour.rs:217]: here also

File: crates/server-config/src/defaults.rs [hunk 3]
```diff
@@ -182,7 +182,7 @@ pub fn default_management_peer_id() -> PeerId {
     let public_key = PublicKey::Ed25519(kp.public());
     let peer_id = PeerId::from(public_key);
 
-    log::warn!(
+    println!(
```
Comment (kmd-fl) [inline crates/server-config/src/defaults.rs:185]: why print and not log/tracing? here and below

File: crates/server-config/src/resolved_config.rs [hunk 4]
```diff
@@ -36,21 +37,58 @@ pub struct UnresolvedConfig {
...
+    Default,
+}
+
+impl FromStr for LogFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.trim().to_ascii_lowercase().as_str() {
+            "logfmt" => Ok(LogFormat::Logfmt),
+            "default" => Ok(LogFormat::Default),
```
Comment (kmd-fl) [inline crates/server-config/src/resolved_config.rs:79]: Is `default` really default? I mean, if I don't pass this flag, will the default format be used?

File: particle-node/src/dispatcher.rs [hunk 5]
```diff
@@ -97,7 +97,7 @@ impl Dispatcher {
                     if let Some(m) = metrics {
                         m.particle_expired(&particle.id);
                     }
-                    log::info!("Particle {} expired", particle.id);
+                    tracing::info!(particle_id = particle.id, "Particle expired");
```
Comment (kmd-fl) [inline particle-node/src/dispatcher.rs:100]: Right below, I see the similar message `Particle is expired`. Mb fix this message or that one? I gather, both variants are correct

File: crates/server-config/src/defaults.rs (hunk 3 above)
Comment (gurinderu) [reply crates/server-config/src/defaults.rs:185]: Because logger isn't available here:(

File: crates/server-config/src/resolved_config.rs (hunk 4 above)
Comment (gurinderu) [reply crates/server-config/src/resolved_config.rs:79]: Yes, it you don't nees to pass arg/env

File: connection-pool/src/behaviour.rs (hunk 2 above)
Comment (gurinderu) [reply connection-pool/src/behaviour.rs]: Fixed

File: aquamarine/src/particle_functions.rs (hunk 1 above)
Comment (gurinderu) [reply aquamarine/src/particle_functions.rs]: Fixed

File: crates/server-config/src/defaults.rs (hunk 3 above)
Comment (kmd-fl) [reply crates/server-config/src/defaults.rs:185]: It's only in the server-config module, right? So, it's called only once? Maybe it's okay, I don't really know

**Review by kmd-fl — APPROVED** [review]

File: crates/particle-node-tests/tests/boolean_test.rs [hunk 6]
```diff
@@ -24,12 +24,14 @@ use serde_json::json;
...
 use connected_client::ConnectedClient;
 use created_swarm::make_swarms;
+use log_utils::enable_logs;
 use service_modules::load_module;
 use test_constants::KAD_TIMEOUT;
 use test_utils::create_service;
 
 #[tokio::test]
 async fn pass_boolean() {
+    enable_logs();
```
Comment (folex) [inline crates/particle-node-tests/tests/boolean_test.rs:34]: Remove that plz

File: particle-node/src/connectivity.rs [hunk 7]
```diff
@@ -85,34 +85,34 @@ impl Connectivity {
                     if let Some(m) = metrics {
                         m.count_resolution(Resolution::ConnectionFailed)
                     }
-                    log::warn!(
-                        "{} Couldn't connect to {} for particle {}",
+                    tracing::warn!(
+                        particle_id = particle_id,
+                        "{} Couldn't connect to {} for particle",
```
Comment (folex) [inline particle-node/src/connectivity.rs:90]:
```suggestion
                        "{} Couldn't connect to {}",
```

File: particle-node/src/connectivity.rs [hunk 8]
```diff
@@ -85,34 +85,34 @@ impl Connectivity {
...
                 }
                 Ok(None) => {
                     if let Some(m) = metrics {
                         m.count_resolution(Resolution::KademliaNotFound)
                     }
-                    log::warn!(
-                        "{} Couldn't discover {} for particle {}",
+                    tracing::warn!(
+                        particle_id = particle_id,
+                        "{} Couldn't discover {} for particle",
```
Comment (folex) [inline particle-node/src/connectivity.rs:101]:
```suggestion
                        "{} Couldn't discover {}",
```

File: particle-node/src/connectivity.rs [hunk 9]
```diff
@@ -85,34 +85,34 @@ impl Connectivity {
...
                 Err(err) => {
                     if let Some(m) = metrics {
                         m.count_resolution(Resolution::KademliaError)
                     }
                     let id = particle_id;
-                    log::warn!(
-                        "{} Failed to discover {} for particle {}: {}",
+                    tracing::warn!(
+                        particle_id = id,
+                        "{} Failed to discover {} for particle: {}",
```
Comment (folex) [inline particle-node/src/connectivity.rs:113]:
```suggestion
                        "{} Failed to discover {}: {}",
```

File: crates/server-config/src/keys.rs [hunk 10]
```diff
@@ -145,7 +144,7 @@ pub fn load_key(
 ) -> eyre::Result<KeyPair> {
     if !key_path.exists() {
         return if generate_on_absence {
-            info!("Generating a new key to {:?}", key_path);
+            println!("Generating a new key to {:?}", key_path);
```
[the remark this answers is not in the document]
Comment (gurinderu) [reply crates/server-config/src/keys.rs:147]: We, can. But I don't want it change a config fully for now. But in future we can make resolving after logger initialization

File: particle-node/src/dispatcher.rs (hunk 5 above)
Comment (folex) [inline particle-node/src/dispatcher.rs:100]:
```suggestion
                    tracing::info!(particle_id = particle.id, "Particle is expired");
```

File: crates/particle-node-tests/tests/boolean_test.rs [hunk 11]
```diff
@@ -24,12 +24,14 @@ use serde_json::json;
 
 use connected_client::ConnectedClient;
 use created_swarm::make_swarms;
+use log_utils::enable_logs;
```
Comment (folex) [inline crates/particle-node-tests/tests/boolean_test.rs:27]:
```suggestion
```

**Review by folex — APPROVED** [review]

**Pull request merged**